### ISSUE TITLE
CI configuration for nightly-build must be fixed

### ISSUE THREAD

**Issue by ruomad** [duckdb/duckdb#8794]
In case it went unnoticed 

Nightly build for Python fails in deploy stage :
```
WARNING  Error during upload. Retry with the --verbose option for more details.
ERROR    HTTPError: 401 Unauthorized from https://upload.pypi.org/legacy/      
         User hfmuehleisen has two factor auth enabled, an API Token or Trusted
         Publisher must be used to upload in place of password.  
```

**Comment by hannes** [comment]
Yes, thanks, there is a PR open to fix this already: #8792

**Comment by szarnyasg** [comment]
@ruomad thanks for reporting this. This belongs to the main DuckDB repository so I transferred it from `duckdb-web` to `duckdb`. The failed job is at https://github.com/duckdb/duckdb/actions/runs/6078737613/job/16490697759
